lib/components/map/connected-transitive-overlay.js

```diff
@@ -1,44 +1,43 @@
-import coreUtils from '@opentripplanner/core-utils'
-import TransitiveCanvasOverlay from '@opentripplanner/transitive-overlay'
+import TransitiveOverlay from '@opentripplanner/transitive-overlay'
+import React, { Component } from 'react'
 import { connect } from 'react-redux'
 
-import { getActiveSearch, getActiveItinerary, getActiveItineraries } from '../../util/state'
+import { getActiveSearch, getTransitiveData } from '../../util/state'
+
+/**
+ * Wrapper for TransitiveOverlay that avoids rerenders by checking transitive
+ * data computed from redux store.
+ */
+class TransitiveCanvasOverlay extends Component {
+  shouldComponentUpdate (nextProps) {
+    return nextProps.transitiveData !== this.props.transitiveData
+  }
+
+  render () {
+    const { transitiveData, ...otherProps } = this.props
+    return (
+      <TransitiveOverlay
+        {...otherProps}
```

Comment:
Let's please try to avoid props spreading.

Reply:
Fixed by #379.